src/plumpy/processes.py

```diff
@@ -1095,22 +1094,29 @@ def pause(self, msg: Union[str, None] = None) -> Union[bool, futures.Cancellable
         if self._stepping:
             # Ask the step function to pause by setting this flag and giving the
             # caller back a future
-            interrupt_exception = process_states.PauseInterruption(msg)
+            interrupt_exception = process_states.PauseInterruption(msg_text)
             self._set_interrupt_action_from_exception(interrupt_exception)
             self._pausing = self._interrupt_action
             # Try to interrupt the state
             self._state.interrupt(interrupt_exception)
             return cast(futures.CancellableAction, self._interrupt_action)
 
-        return self._do_pause(msg)
+        msg = MessageBuilder.pause(msg_text)
+        return self._do_pause(state_msg=msg)
 
-    def _do_pause(self, state_msg: Optional[str], next_state: Optional[process_states.State] = None) -> bool:
+    def _do_pause(self, state_msg: Optional[MessageType], next_state: Optional[process_states.State] = None) -> bool:
         """Carry out the pause procedure, optionally transitioning to the next state first"""
         try:
             if next_state is not None:
                 self.transition_to(next_state)
-            call_with_super_check(self.on_pausing, state_msg)
-            call_with_super_check(self.on_paused, state_msg)
+
+            if state_msg is None:
+                msg_text = ''
+            else:
+                msg_text = state_msg[MESSAGE_KEY]
```

Comment:
these lines are new, why is that

Reply:
Because `_do_pause` in `_create_interrupt_action` is taking a message rather than a msg_text as input. So either I change it there or here accept msg_text.